packages/financial-templates-lib/src/helpers/allowances.js

```diff
@@ -15,6 +15,9 @@ const setAllowance = async (web3, gasEstimator, ownerAddress, spenderAddress, cu
       tx: await collateralToken.methods.approve(spenderAddress, MAX_UINT_VAL).send({
         from: ownerAddress,
         gasPrice: gasEstimator.getCurrentFastPrice(),
+        // Note: Add chainId in case RPC enforces transactions to be replay-protected, (i.e. enforced in geth v1.10,
+        // https://blog.ethereum.org/2021/03/03/geth-v1-10-0/).
+        chainId: await web3.eth.getChainId(),
```

Comment:
is there any way to enforce this on all transactions without needing to add it to every call like this?

Reply:
Yeah good call this is suboptimal. I decided to import the `runTransactions` utility and use it here.

Reply:
Yeah, web3 should definitely be doing this internally IMO. But it doesn't look like they've had any recent releases that address it. I wonder if we could wrap the object in some way to force it to tack this on.

Or maybe there's some sort of property we could set in web3?

https://web3js.readthedocs.io/en/v1.2.11/web3-eth.html#defaultcommon (although the docs aren't super clear on what this actually does...)

Reply:
PTAL as its a bit of sweeping change across different packages